### PR TITLE
Rework Redis trigger docs

### PR DESCRIPTION
**Pull request by itowlson**
Content must go through a pre-merge checklist.

## Pre-Merge Content Checklist

This documentation has been checked to ensure that:

- [x] The `title, `template`, and `date` are all set
- [x] Does this PR have a new menu item (anywhere in `templates/*.hbs` files) that points to a document `.md` that is set to publish in the future? If so please only publish the `.md` and `.hbs` changes in real-time (otherwise there will be a menu item pointing to a `.md` file that does not exist)
- [x] File does not use CRLF, but uses plain LF (hint: use `cat -ve <filename> | grep '^M' | wc -l` and expect 0 as a result) 
- [x] Has passed [`bart check`](https://developer.fermyon.com/bartholomew/quickstart)

Screenshot 2023-03-13 at 12 23 37 pm

- [x] Has run `npm run build-index`
- [x] Has been manually tested by running in Spin/Bartholomew (hint: use `PREVIEW_MODE=1` and run `npm run styles` to update styling)

Screenshot 2023-03-13 at 12 25 10 pm
Screenshot 2023-03-13 at 12 24 44 pm

- [x] Headings are using Title Case
- [x] Code blocks have the programming language set to properly highlight syntax and the proper copy directive
- [x] Have tested with `npm run test` and resolved all errors

Screenshot 2023-03-13 at 12 26 14 pm
Screenshot 2023-03-13 at 12 28 31 pm

### CODE REVIEW

File: content/spin/redis-trigger.md [hunk 1]
```diff
@@ -23,54 +30,154 @@ The Redis instance address is specified in the application trigger:
...
+    println!("{}", from_utf8(&message)?);
+    Ok(())
+}
+```
+
+{{ blockEnd }}
+
+{{ startTab "TypeScript"}}
+
+The JavaScript/TypeScript SDK doesn't currently support Redis components.  Please [let us know](https://github.com/fermyon/spin-js-sdk/issues) is this is important to you.
```
Comment (karthik2804) [inline content/spin/redis-trigger.md:90]:
```suggestion
The JavaScript/TypeScript SDK doesn't currently support Redis components.  Please [let us know](https://github.com/fermyon/spin-js-sdk/issues) if this is important to you.
```

**Review by karthik2804 — COMMENTED** [review]
Just a couple of small suggestions.

File: content/spin/redis-trigger.md (hunk 1 above)
Comment (itowlson) [reply content/spin/redis-trigger.md:90]: Oof, good catch!  Thank you for spotting that!

File: content/spin/redis-trigger.md [hunk 2]
```diff
@@ -1,20 +1,27 @@
...
+- [Specifying an Application as Redis](#specifying-an-application-as-redis)
+- [Mapping a Channel to a Component](#mapping-a-channel-to-a-component)
+- [Redis Components](#redis-components)
+  - [The Message Handler](#the-message-handler)
+- [Inside Redis Components](#inside-redis-components)
 
-Spin applications can be triggered by a new message on a [Redis channel](https://redis.io/topics/pubsub).
-Spin will connect to a configured Redis instance and will invoke components for
-new messages on the configured channels.
+Pub-sub-style messaging is a popular architecture for asynchronous message processing. Spin has built-in support to creating and running applications in response to messages on [pub-sub Redis channels](https://redis.io/topics/pubsub).
```
Comment (fibonacci1729) [inline content/spin/redis-trigger.md:15]: Nit: my attention was strangely drawn to `pub-sub-style` (possibly due to caffeine deficiency); maybe just `pub-sub messaging`?

File: content/spin/redis-trigger.md [hunk 3]
```diff
@@ -23,54 +30,154 @@ The Redis instance address is specified in the application trigger:
...
 
-> We are [exploring adding authentication for connecting to Redis](https://github.com/fermyon/spin/issues/192)
-> and [adding host support for connecting to Redis databases](https://github.com/fermyon/spin/issues/181),
-> from components, which would allow using the key/value store and publishing
-> messages to channels.
+The Redis trigger also requires an `address` field.  This is the address of the Redis instance to monitor.  Specify it using the `redis:` URL scheme.
+
+> If you create an application from a Redis template, the trigger will be already set up for you.
+
+By default, Spin does not authenticate to Redis. You can work around this by providing a password in the `redis://` URL.  For example: `address = "redis://:p4ssw0rd@localhost:6379"`
```
Comment (fibonacci1729) [inline content/spin/redis-trigger.md:37]: Do we need some *ALERT* here to remind users to not commit password to a VCS if they embed them in the address URL like such?

File: content/spin/redis-trigger.md [hunk 4]
```diff
@@ -23,54 +30,154 @@ The Redis instance address is specified in the application trigger:
...
-This is exactly how [the Rust SDK for Spin](./rust-components.md) is built, and,
-as more languages add support for the component model, how we plan to add
-support for them as well.
-
-> We are [exploring a compatibility layer (similar to Wagi)](https://github.com/fermyon/spin/issues/193)
-> so languages that do not have support for the component model can be used to
-> build Redis components for Spin.
+to build a component that Spin can invoke.
+
+This is exactly how Spin SDKs such as the [Rust](rust-components) and [Go](go-components) SDKs are built.
```
Comment (fibonacci1729) [inline content/spin/redis-trigger.md:180]:
```suggestion
This is exactly how Spin SDKs such as the [Rust](rust-components) and [Go](go-components) are built.
```

File: content/spin/redis-trigger.md [hunk 5]
```diff
@@ -23,54 +30,154 @@ The Redis instance address is specified in the application trigger:
...
+As more languages add support for the component model, we plan to add support for them in the same way.
+
+> WIT and the ABI are evolving standards.  The latest version of `wit-bindgen` creates binary implementations that do not work with current language implementation of the WabAssembly System Interface (WASI).  Spin remains pinned to an older implementation of `wit-bindgen` until the next generation of the component model stabilizes and achieves language-level support.
```
Comment (fibonacci1729) [inline content/spin/redis-trigger.md:183]:
```suggestion
> WIT and the ABI are evolving standards.  The latest version of `wit-bindgen` creates binary implementations that do not work with current language implementation of the WebAssembly System Interface (WASI).  Spin remains pinned to an older implementation of `wit-bindgen` until the next generation of the component model stabilizes and achieves language-level support.
```

File: content/spin/redis-trigger.md (hunk 2 above)
Comment (itowlson) [reply content/spin/redis-trigger.md:15]: Good call, also expanded the abbreviation.

File: content/spin/redis-trigger.md (hunk 4 above)
Comment (itowlson) [reply content/spin/redis-trigger.md:180]: Is there a word missing from the suggestion?  "...such as the Rust and Go ones..."?  I'm happier with saying SDKs; if we care deeply about avoiding the repetition then we could change one of them to "libraries" or something.

File: content/spin/redis-trigger.md (hunk 5 above)
Comment (itowlson) [reply content/spin/redis-trigger.md:183]: My plan to create a glorious Wab to overthrow today's creaking Web has been thwarted by one eagle-eyed proofreader.  Thanks for that catch!

**Comment by itowlson** [comment]
Thank you for the careful read @fibonacci1729!

File: content/spin/redis-trigger.md (hunk 4 above)
Comment (fibonacci1729) [reply content/spin/redis-trigger.md:182]: @itowlson just felt like a little stutter. Perhaps a comma would help?

`This is exactly how Spin SDKs, such as the [Rust](rust-components) and [Go](go-components) SDKs, are built.`

OR 

`This is exactly how Spin SDKs are built such as the [Rust](rust-components) and [Go](go-components) SDKs.`

However, free to go with the original; i have sense i'm being a pedant.

Comment (itowlson) [reply content/spin/redis-trigger.md]: I like the commas idea - done and thanks!

File: content/spin/redis-trigger.md [hunk 6]
```diff
@@ -23,54 +30,156 @@ The Redis instance address is specified in the application trigger:
...
-## The WebAssembly Interface
+> Spin subscribes only to the channels that are mapped to components. Other channels are ignored.
+
+## Redis Components
+
+Spin runs Redis components using the [WebAssembly component model](https://github.com/WebAssembly/component-model).  In this model, the Wasm module exports a well-known function that Spin calls to handle the Redis message.
+
+### The Message Handler
+
+The exact signature of the Redis handler, and how a function is identified to be exported as the handler, will depend on your language.
```
Comment (tpmccallum) [inline content/spin/redis-trigger.md:64]:
```suggestion
The exact signature of the Redis handler, and how a function is identified to be exported as the handler, will depend on your language:
```

**Review by tpmccallum — APPROVED** [review]

**Pull request merged**